src/builder.rs

```diff
@@ -0,0 +1,95 @@
+use std::path::{Path, PathBuf};
+
+use crate::Search;
+
+/// Builder for a [`Search`] instance, allowing for more complex searches
+pub struct SearchBuilder {
+    /// The location to search in, defaults to the current directory
+    search_location: PathBuf,
+    /// The search input, defaults to any word
+    search_input: Option<String>,
+    /// The file extension to search for, defaults to any file extension
+    file_ext: Option<String>,
+    /// The depth to search to, defaults to no limit
+    depth: Option<usize>,
+}
+
+impl SearchBuilder {
+    /// Build a new [`Search`] instance
+    #[allow(deprecated)]
+    pub fn build(&self) -> Search {
+        Search::new(
+            &self.search_location,
+            self.search_input.as_deref(),
+            self.file_ext.as_deref(),
+            self.depth,
+        )
+    }
+
+    /// Set the search location to search in
+    /// ### Arguments
+    /// * `location` - The location to search in
+    /// ### Examples
+    /// ```rust
+    /// use rust_search::SearchBuilder;
+    ///
+    /// let search: Vec<String> = SearchBuilder::default().location("src").build().collect();
+    /// ```
+    pub fn location(mut self, location: impl AsRef<Path>) -> Self {
+        self.search_location = location.as_ref().to_path_buf();
+        self
+    }
+
+    /// Set the search input
+    /// ### Arguments
+    /// * `input` - The search input
+    /// ### Examples
+    /// ```rust
+    /// use rust_search::SearchBuilder;
+    ///
+    /// let search: Vec<String> = SearchBuilder::default().input("Search").build().collect();
+    /// ```
+    pub fn input(mut self, input: impl Into<String>) -> Self {
```

Comment:
should we name this `search_input` or `search_text` cause input seems to be a bit generalised...